zerver/webhooks/slack_incoming/view.py

```diff
@@ -25,9 +28,32 @@
 from zerver.lib.webhooks.common import OptionalUserSpecifiedTopicStr, check_send_webhook_message
 from zerver.models import UserProfile
 
+ParamT = ParamSpec("ParamT")
+
+
+def slack_error_handler(view_func: Callable[..., HttpResponse]) -> Callable[..., HttpResponse]:
+    """
+    A decorator that catches JsonableError exceptions and returns a
+    Slack-compatible error response in the format:
+    {ok: false, error: "error message"}.
+    """
+
+    @wraps(view_func)
+    def wrapped_view(
+        request: HttpRequest, *args: ParamT.args, **kwargs: ParamT.kwargs
+    ) -> HttpResponse:
+        try:
+            return view_func(request, *args, **kwargs)
+        except JsonableError as error:
+            error_message = error.msg_format().format(**getattr(error, "data", {}))
+            return JsonResponse({"ok": False, "error": error_message}, status=400)
```

Comment:
We probably want to use the status that's set for the `JsonableError` here and not 400 for all cases.

Also, the error messages that are set for these errors are marked for translation, so maybe it'd be better to send the machine-readable `ErrorCode` enum name as the error message instead.